src/hardware/voodoo.cpp

```diff
@@ -35,154 +38,159 @@
 
 
 class VOODOO;
-static VOODOO* voodoo_dev;
+static std::unique_ptr<VOODOO> voodoo_dev;
+
+constexpr uint32_t voodoo_lfb_mask = 0xffff0000;
+static uint32_t voodoo_current_lfb = (VOODOO_INITIAL_LFB & voodoo_lfb_mask);
 
-static uint32_t voodoo_current_lfb=(VOODOO_INITIAL_LFB&0xffff0000);
+enum class EmulationType {
+    None,
+    Software,
+    OpenGL
+};
+
+enum class CardType : int {
+    Type1 = 1
+};
 
+constexpr const char* NONE_STR = "false";
+constexpr const char* SOFTWARE_STR = "software";
+constexpr const char* OPENGL_STR = "opengl";
+constexpr const char* AUTO_STR = "auto";
 
-class VOODOO:public Module_base{
+class VOODOO : public Module_base {
```

Comment:
Can we make it `Voodoo`?

Reply:
Extending from `Module_base` should not be done, we're trying to get rid of that pattern.

Reply:
Removed the Module_base extension, however didn't change for VOODOO since [changing it causes a bunch of issues on other files... @johnnovak @kcgen

Reply:
As we discussed, please rename the VOODOO class name to Voodoo only (leave the VOODOO_* functions alone).